internal/scheduler/metrics/metrics.go

```diff
@@ -186,11 +186,6 @@ func (m *Metrics) Update(
 			return err
 		}
 	}
-	if jst.Leased {
```

Comment:
I'd add a comment to this function explaining why we're not calling leased here since it may look surprising.